pypeit/pypeit.py

```diff
@@ -858,19 +858,26 @@ def extract_one(self, frames, det, sciImg, objFind, initial_sky, sobjs_obj):
             basename=self.basename)
 
         if not self.par['reduce']['extraction']['skip_extraction']:
-            skymodel, objmodel, ivarmodel, outmask, sobjs, waveImg, \
-                tilts = self.exTract.run(final_global_sky, ra=self.fitstbl["ra"][frames[0]],
-                                         dec=self.fitstbl["dec"][frames[0]], obstime=self.obstime)
+            # Perform the extraction
+            self.exTract.run(final_global_sky)
+            # Retrieve relevant information
+            skymodel = self.exTract.skymodel
+            objmodel = self.exTract.objmodel
+            ivarmodel = self.exTract.ivarmodel
+            outmask = self.exTract.outmask
+            sobjs = self.exTract.sobjs
+            # Apply a reference frame correction to each object and the waveimg
+            self.exTract.refframe_correct(self.fitstbl["ra"][frames[0]], self.fitstbl["dec"][frames[0]], self.obstime,
+                                          sobjs=self.exTract.sobjs)
         else:
-            # Although exrtaction is not performed, still need to prepare some masks and the tilts
-            self.exTract.prepare_extraction()
+            # Although extraction is not performed, still need to prepare some masks and the tilts, and apply heliocentric correction
+            self.exTract.prepare_extraction(final_global_sky)
+            self.exTract.refframe_correct(self.fitstbl["ra"][frames[0]], self.fitstbl["dec"][frames[0]], self.obstime)
```

Comment:
This reframe correct should not modify things in place. Let's get into the habit of returning things from functions so as to not obfuscate control flow and manage scope.